ui/components/component-library/select-wrapper/select-wrapper.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import type { PolymorphicRef, BoxProps } from '../box';
+import { Box, Popover, PopoverPosition } from '..';
+import {
+  SelectWrapperComponent,
+  SelectWrapperProps,
+} from './select-wrapper.types';
+import { SelectContext } from './select-wrapper.context';
+
+export const SelectWrapper: SelectWrapperComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      className = '',
+      defaultValue,
+      value,
+      onValueChange,
+      placeholder,
+      isDanger,
+      isDisabled,
+      isOpen,
+      onOpenChange,
+      isMultiSelect, // Prevents the uncontrolled open state from being toggled
+      triggerComponent,
+      popoverProps,
+      children,
+      onBlur, // Controlled onBlur prop
+      ...props
+    }: SelectWrapperProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    const [uncontrolledValue, setUncontrolledValue] = useState<any | null>();
+    const [isUncontrolledOpen, setIsUncontrolledOpen] =
+      useState<boolean>(false);
+    const [referenceElement, setReferenceElement] =
+      useState<HTMLElement | null>();
+
+    const setBoxRef = (popoverRef: HTMLElement | null) => {
+      setReferenceElement(popoverRef);
+    };
+
+    const toggleUncontrolledOpen = () => {
+      setIsUncontrolledOpen(!isUncontrolledOpen);
+    };
+
+    const handleBlur = (e: any) => {
+      const wrapper = e.currentTarget;
+      const { relatedTarget } = e;
+
+      if (!wrapper.contains(relatedTarget)) {
+        // Close the popover only if the related target is not inside the wrapper
+        setIsUncontrolledOpen(false);
+        // If you have a controlled isOpen state, update it to close
+        if (onOpenChange) {
+          onOpenChange(false);
+        }
+        // Allow the dev to pass in a controlled onBlur prop
+        onBlur?.(e);
+      }
+    };
+
+    return (
+      <SelectContext.Provider
+        value={{
+          isOpen,
+          onOpenChange,
+          isUncontrolledOpen,
+          setIsUncontrolledOpen,
+          toggleUncontrolledOpen,
+          isDisabled,
+          isDanger,
+          defaultValue,
+          value,
+          onValueChange,
+          uncontrolledValue,
+          setUncontrolledValue,
+          placeholder,
+          isMultiSelect,
+        }}
+      >
+        <Box
+          className={classnames('mm-select-wrapper', className)}
+          onBlur={handleBlur}
+          ref={ref}
+          {...(props as BoxProps<C>)}
+        >
+          {triggerComponent &&
+            React.cloneElement(triggerComponent, {
+              ref: setBoxRef,
```

Comment:
This can wait until the `SelectButton` but I think we would pass the entire value object here so what ever the triggerComponent is receives it? That way we may be able to make the SelectButton work outside of the select context.